Parser/tokenizer.h

```diff
@@ -82,8 +82,6 @@ struct tok_state {
     int lineno;         /* Current line number */
     int first_lineno;   /* First line of a single line or multi line string
                            expression (cf. issue 16806) */
-    int fstring_first_constant_lineno; /* First line number of a single line or multiline
```

Comment:
@pablogsal @lysnikolaou I was going to move this to the tokenizer_mode as per your comment on the main PR, but now questioning whether we need it at all or not. Do you have any context on why we might have added this since reusing `first_lineno` works perfectly fine (I've tried it with an assert that checks `fstring_first_constant_lineno==first_lineno` on `token_setup` and for all the examples I can think of this passes including the test suite).

Reply:
Check https://github.com/pablogsal/cpython/pull/46#discussion_r1143421503 (i don't think this collide with that) but we also need to keep the `Python-tokenize.c` up to date with it.

Reply:
Oh, that's a good catch; WDYT about the unification here? https://github.com/pablogsal/cpython/pull/51/commits/bd7483e6a1bf16434290ffbbf468d291da03d4a3

Reply:
I asked @mgmacias95 to review it as she was working on that part to address https://github.com/pablogsal/cpython/pull/46#discussion_r1142809202